### DISCUSSION TITLE
Model used in json result

### DISCUSSION THREAD

**Discussion by SEilers** [openai/whisper#1112] [Ideas]
Could you please include the model (tiny, small, medium.. etc.)  being used during transcription in the json result file?
Background: I have an application where I am caching the results and I would like to have a cache miss, when transcribing the same
file with higher quality.

**Comment by jongwook** [comment]
I'd be open to a PR for this -- meanwhile, you can specify `--output_dir` separately for each model. Alternatively, if you're calling the method in Python, you can simply add the model name in the returned dictionary.

**Reply by ururk** [reply]
PR submitted: https://github.com/openai/whisper/pull/1225

I'm not sure what parameters would be most useful for everyone, so I submitted the PR for just the model name, but I presume others params could be added hence the nested dict.

**Reply by ururk** [reply]
Also, not happy about having to pass in the name of the model to `transcribe()` but I didn't see a way to get the name of the model from the loaded model.

**Comment by ururk** [comment]
Yes, this is a good idea! I wrote a python script to call Whisper to do just this to help me benchmark / analyze different runs. I add a `job_details` key to the dictionary and output the whole thing as JSON:

```JSON
  "job_details": {
    "program": "whisper",
    "file": "/absolute/path/to/file.mp4",
    "model": "medium.en",
    "threads": 0,
    "device": "cpu",
    "condition_on_previous_text": true,
    "processing_time": 1794.0216725
  },
```

Might be good for it to capture all params for a run? These are the only params I was interested in, but other params will obviously impact things like run time and acuracy.

**Reply by dgoryeo** [reply]
Hi @ururk , this looks quite helpful for testing. Would you be able to share your python code here?

**Reply by ururk** [reply]
Sure - here's what I'm using. It has little to no error checking (mostly none), more or less is a clone of the `cli` def in `transcribe.py` with some extra bits added, and some `pyinstaller`-specific things so I can bundle it (and faster-whisper) into a single folder+executable that is then portable across my fleet of machines. I also throw in `jiwer` to calculate WER against a provided transcript. Yes - this could be done post-whisper, or as a separate routine (ie, broken out into a different module) but I'm not going for optimal code here, just something that works so I can validate using Whisper.

https://gist.github.com/ururk/51425f1482c3a9f59e23074acb61bede
https://gist.github.com/ururk/e07198754dcb1ce3adde27f664609183

I'll start a pull request with changes to transcribe.py.

**Reply by dgoryeo** [reply]
Thanks @ururk
